src/content/en/fundamentals/media/mobile-web-video-playback.md

```diff
@@ -0,0 +1,611 @@
+      <div id="videoControls"></div>
+    </div>
+
+### Read video metadata
+
+First, let's wait for the video metadata to be loaded to set the video
+duration, the current time, and initialize the progress bar. Note that the
+`secondsToTimeCode()` function is a custom utility function I've written that
+converts a number of seconds to a string in "hh:mm:ss" format which is better
+suited in our case.
+
+<pre class="prettyprint lang-html">
+&lt;div id="videoContainer">
+  &lt;video id="video" src="file.mp4">&lt;/video&gt;
+  &lt;div id="videoControls">
+    <strong>&lt;div id="videoCurrentTime">&lt;/div>
+    &lt;div id="videoDuration">&lt;/div>
+    &lt;div id="videoProgressBar">&lt;/div></strong>
+  &lt;/div>
+&lt;/div>
+</pre>
+
+    video.addEventListener('loadedmetadata', function() {
+      videoDuration.textContent = secondsToTimeCode(video.duration);
+      videoCurrentTime.textContent = secondsToTimeCode(video.currentTime);
+      videoProgressBar.style.transform = `scaleX(${video.currentTime / video.duration})`;
+    });
+
+<figure>
+  <img src="/web/fundamentals/media/images/video-metadata-only.png">
+  <figcaption>
+    <b>Figure 2.</b> Media Player showing video metadata
+  </figcaption>
+</figure>
+
+### Play/pause video
+
+Now that video metadata are loaded, let's add our first button that lets user
+play and pause video with `video.play()` and `video.pause()` depending on its
+playback state.
+
+<pre class="prettyprint lang-html">
+&lt;div id="videoContainer">
+  &lt;video id="video" src="file.mp4">&lt;/video&gt;
+  &lt;div id="videoControls">
+    <strong>&lt;button id="playPauseButton">&lt;/button></strong>
+    &lt;div id="videoCurrentTime">&lt;/div>
+    &lt;div id="videoDuration">&lt;/div>
+    &lt;div id="videoProgressBar">&lt;/div>
+  &lt;/div>
+&lt;/div>
+</pre>
+
+    playPauseButton.addEventListener('click', function(event) {
+      event.stopPropagation();
+      if (video.paused) {
+        video.play();
+      } else {
+        video.pause();
+      }
+    });
+
+Note: I call `event.stopPropagation()` to prevent parent handlers (e.g. video
+controls) from being notified of the click event.
+
+Rather than adjusting our video controls in the `click` event listener, we use
+the `play` and `pause` video events. When video starts playing, we change the
+button state to "pause" and hide the video controls. When the video pauses, we
+simply change button state to "play" and show the video controls.
+
+    video.addEventListener('play', function() {
+      playPauseButton.classList.add('paused');
+    });
+
+    video.addEventListener('pause', function() {
+      playPauseButton.classList.remove('paused');
+    });
+
+When time indicated by video `currentTime` attribute changed via the
+`timeupdate` video event, we also update our custom controls if they're
+visible.
+
+    video.addEventListener('timeupdate', function() {
+      if (videoControls.classList.contains('visible')) {
```

Comment:
It is a plain class I'm using to indicate that video controls are visible or not.
See explanation text above.